src/RestaurantApi/Users/EditUser.js

```diff
@@ -0,0 +1,31 @@
+import { auth, db } from "../../firebase"
+import { updateDoc, doc} from 'firebase/firestore'
+import { updateEmail, updatePassword } from "firebase/auth";
+
+export const editUser = async (id, email, password, name) => {
+  const userDoc = doc(db, "users", id);
+  const user = auth.currentUser;
+
+  const newFields = {
+    email: email, 
+    name: name, 
+  };
+
+  try {
+    await updateDoc(userDoc, newFields);
+  } catch (e) {
+    console.log(e);
+  }
+
+  try {
+    await updateEmail(user, email);
+  } catch (e) {
+    console.log(e);
+  }
+
+  try {
+    await updatePassword(user, password);
+  } catch (e) {
+    console.log(e);
+  }
```

Comment:
these three operations should be written in transaction because it is possibility to updateDoc succesfully, updateEmail succesfully but updatePassword can fail. In that case all changes have to rollback. Try find something about transactions in firebase docs: https://firebase.google.com/docs/firestore/manage-data/transactions and please try to implement this or just make this one update method for user doc.